### PR TITLE
MISUV-3671: update the order of entries in PaymentHistory

### PR DESCRIPTION
**Pull request by avelx**
* update the order of entries in PaymentHistory: earliest first order

### CODE REVIEW

File: app/views/PaymentHistory.scala.html [hunk 1]
```diff
@@ -42,12 +42,21 @@
...
+                .map(_._2)
+}
+
 @groupedPayments = @{
     payments.groupBy[Int]{payment => {
             if (payment.validMFACreditDescription()) LocalDate.parse(payment.documentDate).getYear
             else LocalDate.parse(payment.dueDate.getOrElse(throw MissingFieldException("Payment Date"))).getYear
         }
     }.toList.sortBy(_._1).reverse
+      .map(rec => (rec._1, sortPayments(rec._2)))
```
Comment (joesheridan) [inline app/views/PaymentHistory.scala.html:59]: Might be nice to use case match here to give the tuple params some context:
.map { case(year, payments) => (year, sortPayments(payments)) }

File: app/views/PaymentHistory.scala.html [hunk 2]
```diff
@@ -42,12 +42,21 @@
 btaNavPartial: Option[Html] = None, origin: Option[String] = None, MFACreditsEnabled: Boolean = true)(implicit request: Request[_], messages: Messages)
 @import implicitDateFormatter.longDate
 
+@sortPayments(payments: List[Payment]) = @{
+            payments
```
Comment (joesheridan) [inline app/views/PaymentHistory.scala.html:46]: spacing seems a bit off - tabs should be 4 spaces in template files

File: app/views/PaymentHistory.scala.html [hunk 3]
```diff
@@ -42,12 +42,21 @@
...
+                .map(payment => if (payment.validMFACreditDescription() ) ( Some(payment.documentDate), payment) else (payment.dueDate, payment) )
+                .map(rec => ( LocalDate.parse(rec._1.getOrElse(throw MissingFieldException("Payment/Credits::Sorting"))).toEpochDay, rec._2) )
```
Comment (joesheridan) [inline app/views/PaymentHistory.scala.html:48]: case match with map, perhaps?

File: test/views/PaymentHistoryViewSpec.scala [hunk 4]
```diff
@@ -57,12 +60,18 @@ class PaymentHistoryViewSpec extends ViewSpec with ImplicitDateFormatter {
...
 
   val testPayments: List[Payment] = List(
-    Payment(Some("AAAAA"), Some(10000), None, Some("Payment"), None, Some("lot"), Some("lotitem"), Some("2019-12-25"),
+    Payment(Some("AAAAA"), Some(10000), None, Some("Payment"), None, Some("lot"), Some("lotitem"), Some("2019-11-25"),
+      "2019-11-25", Some("DOCID01")),
+    Payment(Some("AAAAB"), Some(10000), None, Some("Payment"), None, Some("lot"), Some("lotitem"), Some("2019-12-25"),
       "2019-12-25", Some("DOCID01")),
-    Payment(Some("BBBBB"), Some(5000), None, Some("tnemyap"), None, Some("lot"), Some("lotitem"), Some("2007-03-23"),
-      "2019-12-25", Some("DOCID02"))
+    Payment(Some("BBBB1"), Some(5000), None, Some("tnemyap"), None, Some("lot"), Some("lotitem"), Some("2007-03-23"),
```
Comment (joesheridan) [inline test/views/PaymentHistoryViewSpec.scala:67]: We should either add some MFA credits in this list or adjust the paymentsMFA to check that the sorting is working for MFA

Comment (avelx) [reply test/views/PaymentHistoryViewSpec.scala:67]: * paymentsMFA test adjusted to verify sorting order is correct;

**Review by joesheridan — APPROVED** [review]

**Pull request merged**